packages/p2p/src/components/orders/order-details/order-details-footer.jsx

```diff
@@ -1,198 +1,368 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Button } from '@deriv/components';
-import { localize, Localize } from 'Components/i18next';
+import { Button, Checkbox, Modal, RadioGroup } from '@deriv/components';
+import { useIsMounted } from '@deriv/shared';
 import { requestWS } from 'Utils/websocket';
-import Dp2pContext from 'Components/context/dp2p-context';
+import { localize, Localize } from 'Components/i18next';
+import FormError from '../../form/error.jsx';
+import './order-details.scss';
+
+const CancelOrderModal = ({ id, hideCancelOrderModal, should_show_cancel_modal }) => {
+    const isMounted = useIsMounted();
+    const [error_message, setErrorMessage] = React.useState('');
 
-const OrderDetailsFooter = ({ order_information, cancelPopup, showPopup }) => {
-    const { email_domain } = React.useContext(Dp2pContext);
+    const cancelOrderRequest = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_order_cancel: 1,
+                id,
+            }).then(response => {
+                if (isMounted()) {
+                    if (response.error) {
+                        setErrorMessage(response.error.message);
+                    }
+                }
+                resolve();
+            });
```

Comment:
We can remove the `Promise` wrapped around the request as we're not waiting for this function to resolve.